branch-name-content-script.js

```diff
@@ -1,6 +1,7 @@
 if (window === top) {
   chrome.runtime.onMessage.addListener(function(message, sender, sendResponse) {
-    sendResponse(getBranchName());
+    if (message.addButton) addBranchNameButton();
+    else sendResponse(getBranchName());
```

Comment:
Trying to remember how all this stuff works, so looking for education here more than changes:

- Is there a case when message.addButton will not be true? Is sending a message with addButton: true necessary, or could addBranchNameButton on L3 be called directly without a gate?

Reply:
They probably can be condensed,  the life cycle hooks of this were not obvious to me.  According to the docs there are a couple of paths you can go down for knowing when navigation within a page occurs and I'd implemented some stuff using a different chrome api along those lines that I ripped out because the tabs and it both worked.  I'll test if I can just have both these actions on the same message.

Reply:
Merged the two messages and it's working fine.